src/JsonSchema/Constraints/Constraint.php

```diff
@@ -28,8 +28,9 @@
     protected $errors = array();
     protected $inlineSchemaProperty = '$schema';
 
-    const CHECK_MODE_NORMAL = 1;
-    const CHECK_MODE_TYPE_CAST = 2;
+    const CHECK_MODE_NORMAL = 0x00000001;
```

Comment:
Just a style thing I learned decades ago. I guess the idea is that this way it's a little more clear to the reader that we're creating a sequence for the purpose of bit-masking.

```
0x00000001
0x00000002
0x00000004
0x00000008
0x00000010
0x00000020
0x00000040
```
vs

```
1
2
4
8
16
32
64
```
But I could go either way if you feel strongly about it.